### ISSUE TITLE
Improve limits panel to account for partial liquidation

### ISSUE THREAD

**Issue by valiafetisov** [sidestream-tech/unified-auctions-ui#478]
### Goal

Limits panel do not prevent partial execution

### Context

Currently, the vault ID proposed as an example in the production `/vaults` page is presented with `VaultLiquidationLimitsCheckPanel` showing an error, while in fact it's partial execution is possible, so it should be presented as warning, not error (and not pass error state up in the `isCorrect` prop).

The best way to determine (on the frontend) if partial execution is possible is to be suggested by @KirillDogadin-std. Since it's a complex calculation involving checking if the part that are taken or the part that is left out is big enough, I suggest it will be derived from some vault values, rather than recomputed on the frontend.

Suggested changes in this PR:
- [ ] Highlight reached limits (-23364594.25 DAI in the screenshot below) in orange for partial liquidation (or red for no liquidation)
- [ ] Change color and content of the title accordingly (red or orange)

### Assets

<img width="666" alt="Screenshot 2022-09-26 at 17 51 26" src="https://user-images.githubusercontent.com/3393626/192325384-592ddce5-f655-45bc-afe0-e0f1e3a1ab28.png">

### Tasks

- [ ] Improve visibility on which limits are reached
- [ ] Improve signalling of partial execution via title and color
- [ ] Unblock partial execution

**Comment by KirillDogadin-std** [comment]
> The best way to determine (on the frontend) if partial execution is possible

"on the frontend" is the important part here. 

You have proposed the logic that uses math and i don't really think it's appropriate because:
  - math on "frontend" is a thing to avoid : we should do all the necessary computations in the "backend". (quotes are used here since it's a d-app. We instead have "ui" and "core")
  - all these computations are already done on the blockchain and hence should be done in the core so that it's sufficient information to answer all the questions.

To my current knowledge the following should actually suffice:
   - if the auctions is in the `liquidatable` state while this 'liquidation limits are reached' __warning__ is displayed: the partial liquidation is possible.
   - liquidation limits are reached when `maxProtocolDebt - currentProtocolDebt - incentivesForStartingAuction < 0` (variable names are made up in this bullet point, don't CMD+F/^F them)

> How should we scale the incentives to avoid [farming](https://docs.makerdao.com/smart-contract-modules/dog-and-clipper-detailed-documentation#incentive-farming)?

in the linked part i see the following: 

> governance must take care when increasing

which means that we don't have to do anything with it since we're not "governance". Aka only big guys with admin rights can overwrite constants in the protocol - we are not these guys.

**Comment by KirillDogadin-std** [comment]
> If a user decides to liquidate just 100 of those 1000 debt DAI, then they would get the full constant incentive (100 DAI) and 10 DAI from the relative incentive. Is my understanding correct?

can't answer question because the axiom in it is incorrect:

"If a user decides" - wrong. User can't decide at all. User can only ask to liquidate the vault. How much of the vault's collateral is liquidated to cover the debt is not up to the user, but up to the protocol.

___

I've taken a closer look. Please note the following bullet points. Sorry if i'm introducing more confusion now, but actually we might even discover some other bugs here. Please try to understand every bullet point and ask questions if not clear.

  - there're 5 parameters in the auctioning process: incentive relative, incentive absolute, debt, liquidation penalty, auctioned dai
  - stage __before__ auction created on chain - the relevant parameters are: debt, liquidation penalty
    - debt and liquidation penalty[ form together ](https://github.com/makerdao/dss/blob/fa4f6630afb0624d04a003e920b0d71a00331d98/src/dog.sol#L224)the value of the auctioned dai
  - stage __during__ auction being created on chain - the relevant parameters are: both incenitves, auctioned dai
    - incentives define [how much dai](https://github.com/makerdao/dss/blob/60690042965500992490f695cf259256cc94c140/src/clip.sol#L261) will be sent out to the keeper (depends on the auctioned dai)
___

Giving example now.

Debt of the vault: 1000 DAI
Currently auctioned dai: 1300 DAI
Maximum auctioned dai: 2000 DAI
The liquidation penalty: 10%
The minimal auctioned/contained debt: 100 DAI
Incentives: 10 Absolute, 10% relative

- we can only auction 2000 - 1300 = 700 DAI while the vault owes 1000.
- We can partially liquidate the vault and auction 700 dai. But this value includes the liquidation penalty. So the debt that will be covered is 700 * (1 - 0.1) = 630 dai. because the liquidation penalty is 10%.
- the leftover of the debt is going to be 470 Dai. This is greater than 100 (minimal value), the auctioned dai is 630 and this is also greater than 100. So we're good on minimal limit there
- we will open the auction now. This auction will cover 630 dai of debt but will start with price tag of 700 dai because of the liquidation penalty.
- the incentives that will be sent to the guy who opened the auction: 700 * 0.1 + 10 = 80 dai

___

if the above is clear please check

- [ ] frontend uses `maximumProtocolDebt - currentProtocolDebt - debt * liquidationPenalty < 0` to show the warning. As far as i understand the current state is that it subtracts debt and incentives.

**Comment by KirillDogadin-std** [comment]
> why aren't the incentives included in the calculation? 

that's up to the protocol engineers, i am just saying how it is. Let's not speculate on the intents there. But seems logical that these are decoupled - they're parts of 2 different contracts.
